src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -348,13 +351,19 @@ private GoogleBucketResource retrieveBucketBy(String sql, MapSqlParameterSource
         List<GoogleBucketResource> bucketResources =
             jdbcTemplate.query(sql, params, (rs, rowNum) -> {
                 // Make project resource and a bucket resource from the query result
+                UUID bucketResourceId = rs.getObject("project_resource_id", UUID.class);
                 GoogleProjectResource projectResource = new GoogleProjectResource()
-                    .id(rs.getObject("project_resource_id", UUID.class))
+                    .id(bucketResourceId)
                     .googleProjectId(rs.getString("google_project_id"))
                     .googleProjectNumber(rs.getString("google_project_number"))
                     .profileId(rs.getObject("profile_id", UUID.class));
 
-                String region = rs.getString("region");
+                String region;
+                try {
+                    region = storageDao.getBucketStorageFromBucketResourceId(bucketResourceId);
+                } catch (StorageResourceNotFoundException e) {
```

Comment:
The region returned by `storageDao.getBucketStorageFromBucketResourceId` will never be null. It will throw a `StorageResourceNotFound` exception if we don't have a region for the bucket in the DB